lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1256,6 +1256,7 @@ def deploy_vm(self):
                 if self.params['wait_for_ip_address']:
                     self.wait_for_vm_ip(vm)
 
+            self.change_detected = True
```

Comment:
i prefer having it near the change owner, it's more easy to maintain if we want to do more things on role